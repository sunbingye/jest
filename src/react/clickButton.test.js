import renderer from 'react-test-renderer';
import ClickButton from './clickButton';

test('changes the class when hovered', () => {
    const component = renderer.create(<ClickButton />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
console.log(`tree:`,tree);
    // 手动出发 button1 点击事件
    renderer.act(() => {
        tree[1].props.onClick();
    });
    // 重绘
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // 手动出发 button2 点击事件
    renderer.act(() => {
        tree[2].props.onClick();
    });
    // 重绘
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
