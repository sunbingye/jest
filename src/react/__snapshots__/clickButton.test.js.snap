// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
[
  <label>x
    nothing was clicked
  </label>,
  <button
    onClick={[Function]}
  >
    button1
  </button>,
  <button
    onClick={[Function]}
  >
    button2
  </button>,
]
`;

exports[`changes the class when hovered 2`] = `
[
  <label>
    button1 was clicked
  </label>,
  <button
    onClick={[Function]}
  >
    button1
  </button>,
  <button
    onClick={[Function]}
  >
    button2
  </button>,
]
`;

exports[`changes the class when hovered 3`] = `
[
  <label>
    button2 was clicked
  </label>,
  <button
    onClick={[Function]}
  >
    button1
  </button>,
  <button
    onClick={[Function]}
  >
    button2
  </button>,
]
`;
